### PR TITLE
resolved consistency error for merged indexer xml files

### LINKED ISSUE #29609: [Issue] resolved consistency error for merged indexer xml files
This issue is automatically created based on existing pull request: magento/magento2#27212: resolved consistency error for merged indexer xml files

---------
The allowed values in classType should be equal in unmerged and merged indexer.xml files

Allowing only letters and backspaces is wrong, since PHP-namespaces can also contain numbers.

### Preconditions: (*)

Magento 2.4-develop

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Steps to Reproduce: (*)

1. Create a simple module with a vendor name which contains at least one number (ex. Cus2tom2)
2. Add a custom indexer to the module just like on the [devdocs](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/indexing-custom.html)
3. Run in console: `php bin/magento setup:upgrade;`

**Actual Result:** :heavy_multiplication_x: Invalid doc element "indexer" , attribute "class" ...

Screenshot from 2020-09-29 14-02-19

**Expected Result:** :heavy_check_mark: Normal Upgrade output & module gets properly installed.

Screenshot from 2020-09-29 14-07-26

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by philippsander**
The allowed values in classType should be equal in unmerged and merged indexer.xml files

Allowing only letters and backspaces is wrong, since PHP-namespaces can also contain numbers.

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#29609: resolved consistency error for merged indexer xml files

### CODE REVIEW

File: lib/internal/Magento/Framework/Indexer/etc/indexer_merged.xsd [hunk 1]
```diff
@@ -57,11 +57,11 @@
     <xs:simpleType name="classType">
         <xs:annotation>
             <xs:documentation>
-                Class name can contain only [a-zA-Z\].
+                Class name can contain only [a-zA-Z|\\]+[a-zA-Z0-9\\]+.
             </xs:documentation>
         </xs:annotation>
         <xs:restriction base="xs:string">
-            <xs:pattern value="[a-zA-Z\\]+" />
+            <xs:pattern value="[a-zA-Z|\\]+[a-zA-Z0-9\\]+"/>
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/Indexer/etc/indexer_merged.xsd:64]: Is `\9ThClass` allowed?
https://regex101.com/r/bu1w2N/1/

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]

**Comment by philippsander** [comment]
@lbajsarowicz If you change it here you should also change the unmerged files

**Comment by philippsander** [comment]
@lbajsarowicz 
I honestly don't see what update is needed here.

The pipeline will never be green since it's a global magento2 issue on ALL pull-requests

**Comment by lbajsarowicz** [comment]
In my opinion, the regex should be: https://regex101.com/r/bu1w2N/2

**Comment by philippsander** [comment]
@lbajsarowicz I agree, but with my merge request I was just trying to fix consitency error.

changing the regex would be too drastic

**Comment by lbajsarowicz** [comment]
Actually I sat one longer moment to find out what's the right approach to this regex and we can even extend it:
https://regex101.com/r/bu1w2N/3

**Comment by sidolov** [comment]
@magento create issue

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-8038](https://jira.corp.magento.com/browse/ENGCOM-8038) has been created to process this Pull Request
:eight_spoked_asterisk: @sidolov, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by philippsander** [comment]
I don't get how I can push this forward. The checks won't execute.

**Review by philippsander — COMMENTED** [review]
Works

**Review by lbajsarowicz — APPROVED** [review]
✔️  Green from my side, although the change is not enough in my opinion.

**Comment by philippsander** [comment]
The checks won't execute.
When will this be merged?

**Comment by lbajsarowicz** [comment]
@magento run all tests.

**Comment by engcom-Alfa** [comment]
Hi @lbajsarowicz @sidolov .
Could you put an appropriate label for test coverage?
Thanks!

**Comment by lbajsarowicz** [comment]
@magento run all tests

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Create a simple module with a vendor name which contains at least one number (ex. Cus2tom2)
2. Add a custom indexer to the module just like on the [devdocs](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/indexing-custom.html)
3. Run in console: `php bin/magento setup:upgrade;`

**Before:** :heavy_multiplication_x: Invalid doc element "indexer" , attribute "class" ...

Screenshot from 2020-09-29 14-02-19

**After:** :heavy_check_mark: Normal Upgrade output & module gets properly installed.

Screenshot from 2020-09-29 14-07-26

**Comment by engcom-Golf** [comment]
@magento run all tests

**Comment by engcom-Golf** [comment]
@magento run all tests

**Comment by engcom-Golf** [comment]
@magento run all tests

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-8038](https://jira.corp.magento.com/browse/ENGCOM-8038) has been created to process this Pull Request

**Pull request merged**